python/packages/nisar/workflows/stage_watermask.py

```
@@ -367,18 +383,101 @@ def download_watermask(polys, epsgs, outfile, version):
-    except Exception:
+        vrt_dataset = gdal.BuildVRT(outfile, watermask_list)
+
+        # Get the WATER description from the LICENSE.txt file using GDAL
+        in_license_path = vrt_filename.replace(f'EPSG{epsg}.vrt',
+                                               f'EPSG{epsg}/LICENSE.txt')
+        license_exists = gdal.VSIStatL(in_license_path) is not None
+
+        if license_exists:
+            try:
+                water_descr = parse_license_notes(in_license_path)
+                vrt_dataset.SetMetadataItem("water_mask_description",
+                                            f'{water_descr}')
+
+            except Exception as e:
+                warnings.warn(
+                    f"Found {in_license_path} but could not parse it ({e}). "
+                    "Proceeding without metadata."
+                )
+        else:
+            warnings.warn(
+                f"No LICENSE.txt found at {in_license_path} "
+                f"(expected for WATERMASK v{version}). Proceeding without metadata."
+            )
+    except Exception as e:
         errmsg = f'Failed to download NISAR WATERMASK {version} from s3 bucket. ' \
                  f'Maybe {version} is not currently supported.'
-        raise ValueError(errmsg)
+        raise ValueError(errmsg) from e
+
+
+def parse_license_notes(license_path: str,
+                        encoding: str = "utf-8") -> str | None:
+    """
+    Extract the multi-line text of the “Notes” bullet from
+    a LICENSE.txt in an S3 bucket via GDAL and returning the
+    value without the leading "- Notes:" header.
+
+    Parameters
+    ----------
+    license_path : str
+        `/vsis3/.../LICENSE.txt` or any other GDAL compatible URI.
+    encoding : str, default "utf-8"
+        Text encoding to decode the bytes.
+
+    Returns
+    -------
+    str | None
+        The extracted Notes text (possibly multi-line) with surrounding
+        whitespace stripped, or None if no “Notes” bullet is found.
```

Let's raise an exception if no "Notes" bullet was found, instead of returning `None`. In that case, the calling code will catch the exception, emit a warning, and proceed without metadata.